enhancements/authentication/configuring-webhook-token-authenticators.md

```diff
@@ -0,0 +1,159 @@
+     to distinguish our "human" users from users authenticated otherwise, so that these
+     are capable of creating their own projects. Would it be ok to add this group to
+     anyone authenticated via a webhookTokenAuthenticator?
+
+## Summary
+
+`webhookTokenAuthenticators` configuration of the kube-apiserver will be configurable and will be leveraged by
+the cluster-authentication-operator to set up cluster authentication with the integrated osin-based OAuth server.
+
+## Motivation
+
+The `webhookTokenAuthenticators` APIServer configuration, along with `oauthMetadata`, played an important role
+for integration with the Keycloak server as a replacement for the integrated osin-based OAuth server. The
+ability to configure the `webhookTokenAuthenticators` for kube-apiserver was lost in the transition from
+3.11 to 4.1, and so the integration with Keycloak is also not possible.
+
+`webhookTokenAuthenticators` could be used for configuring a custom OAuth server which supports Kubernetes'
+token reviews (see [Webhook Token Authentication](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#webhook-token-authentication)
+for more details). Therefore, by implementing this enhancement, we would significantly simplify any
+drop-in replacements for the integrated oauth-server, which might be helpful in cases where we would
+like to delegate authentication to a different OpenShift cluster, as an example.
+
+### Goals
+
+1. Make it possible to configure an endpoint capable of performing kube token reviews
+2. Modify cluster-authentication-operator so that it configures authentication via the OpenShift OAuth
+   server by using the `webhookTokenAuthneticators` field instead of the current patching of the
+   kube-apiserver.
+
+## Proposal
+
+### Implementation
+
+#### API
+
+`authentications.config.openshift.io/v1` already contains the `spec.webhookTokenAuthenticators` field
+that expects an array of references to secrets in the `openshift-config` namespace. This field could
+be used for the purposes of configuring webhook token authenticators, where we would just expect references
+to secrets, that would have to be checked.
+
+The configuration mentioned above leaves quite a lot of freedom in hands of users as the referenced secrets
+would be quite general kubeconfigs which makes users be able to configure fields that could make the
+kube-apiserver access user-specified paths, which may not be desirable (although could be ruled out
+by validation).
+
+An alternative is to handle the configuration by introducing a new cluster-wide CRD:
+```
+apiVersion: config.openshift.io/v1
+kind: TokenAuthenticator
+spec:
+  ca: CM reference
+  server: string (valid https:// URL)
+  clientCertificate: CM reference
+  clientKey: Secret reference
+```
+
+#### Backend
+
+__cluster-authentication-operator__
```

Comment:
this whole section looks incorrect. the only job of this operator is to install its own webhooktokenauthenticator.  It won't even be installed on all clusters in the end. The operator was straight misnamed.

Reply:
I moved the steps that made sense to KAS-o section, hopefully that's better now